### DISCUSSION TITLE
Add osu!lazer to flathub

### DISCUSSION THREAD

**Discussion by BiteDasher** [ppy/osu#15278] [Ideas] [answered]
I'd like to submit osu!lazer for inclusion in Flathub, to make it more easily available on Linux distributions. Though the AppImage is helpful, they're generally not discoverable through desktop environment's built-in app stores like GNOME Software or KDE Discover.

Currently I opened PR in https://github.com/flathub/flathub/pull/2599 and would like to ask @peppy agreement for changing PR status

I want to use sh.ppy.osulazer application ID 

And

Publish application on flathub

**Comment by bdach** [comment]
please do not. the appimage is purposefully the only official distribution method for linux. we've had issues with unaffiliated third parties adding lazer to a multitude of other package managers and then end users coming to us to complain about those non-appimage releases when we've never actually intended to support them in the first place.

see also:
https://github.com/ppy/osu/issues/2960
https://github.com/ppy/osu/issues/7840
https://github.com/ppy/osu/issues/3471#issuecomment-423773506
https://github.com/ppy/osu/issues/3727
https://github.com/NixOS/nixpkgs/issues/132610

**Reply by BiteDasher** [reply]
#7840 was updated a year ago, today I tested the game in AppImage on Fedora 34 and it just wouldn't start and give some errors, so I decided to build the flatpak version.  I wrote a manifest, which can be viewed at the pull request link, and everything works quite well without patches, the music is playing, the map editor, the network, and all game modes are working.  Maybe all the same it is worth considering this option at least as unofficial?

**Reply by BiteDasher** [reply]
It's just that I haven't seen any activity lately regarding testing the game in a flatpak.

**Reply by bdach** [reply]
no offence, but my first reaction to the game not launching would be to open a discussion about the crash (with relevant information & logs included), rather than to repackage the entire thing and add it to another package manager.

**Comment by peppy** [comment]
This seems super weird. If we were wanting to submit osu! to what looks to be an app store front-end, it would be done by us directly.

For now I'd ask that you refrain from this and use the provided distribution formats.

**Comment by Schweeeeeeeeeeeeeeee** [comment]
So you dont want to officially release a flatpak version?
If yes why not i didnt uderstand why?

**Reply by bdach** [reply]
the same reason why we don't want to officially release and maintain an AUR image, an ebuild, a nix flake..... too. you get the idea. linux has too many distribution channels that we can't control (and we *must* control them to ensure game integrity).

stuff has changed here and we sorta caved in, and https://github.com/flathub/sh.ppy.osu/ exists now, but it is still not to be considered official i guess

**Reply by peppy** [reply, marked as answer]
I'm kinda in control of that package now, so it's pretty much official.

**Reply by deviantsemicolon** [reply]
> the same reason why we don't want to officially release and maintain an AUR image, an ebuild, a nix flake..... too. you get the idea. linux has too many distribution channels that we can't control (and we _must_ control them to ensure game integrity).
> 
> stuff has changed here and we sorta caved in, and https://github.com/flathub/sh.ppy.osu/ exists now, but it is still not to be considered official i guess

Regarding the nix thing, there is a package in nixpkgs that builds the app image version called osu-lazer-bin. It isn't official, but it does work

**Reply by deviantsemicolon** [reply]
> > the same reason why we don't want to officially release and maintain an AUR image, an ebuild, a nix flake..... too. you get the idea. linux has too many distribution channels that we can't control (and we _must_ control them to ensure game integrity).
> > stuff has changed here and we sorta caved in, and https://github.com/flathub/sh.ppy.osu/ exists now, but it is still not to be considered official i guess
> 
> Regarding the nix thing, there is a package in nixpkgs that builds the app image version called osu-lazer-bin. It isn't official, but it does work

adding onto this, the AUR also has a pkgbuild which builds the appimage version that shares the same name as the nix package. if anyone plans to port osu-lazer to their own package manager, i'd recommend just building the appimage

**Reply by ghost** [reply]
> I'm kinda in control of that package now, so it's pretty much official.

Why is it unverified?

**Reply by xdlolrgb** [reply]
> I'm kinda in control of that package now, so it's pretty much official.

if its "official" why not put the flatpak as one of the linux download options? appimage is a bit of a meme that still doesnt work on a number of distros (the number used to be higher because the appimage dev cant be arsed to make use of non-EOL dependencies) and cant self update which are two pretty huge pain points for osu lazer. if you want universal distro-agnostic packaging flatpak is pretty much usually superior and much more well supported

**Reply by SupDos** [reply]
> and cant self update

The appimage can self update

**Reply by xdlolrgb** [reply]
It still has a plethora of other issues, which I listed. Flatpaks on Flathub is just a superior application distribution method for the majority of users. If SteamOS ever takes off (it probably won't but Valve is trying to bring it to more platforms besides the handheld) then Flathub will be basically the main way people get software on that (and other immutable distros). I guess you can still use the AppImage but my experience with AppImages on the Steam Deck is nothing sort of clunky, making it integrate with the system outside of desktop mode and all

**Reply by xdlolrgb** [reply]
But that's kind of a minor problem that the flatpak isn't listed among the main download options so it's probably fine. Still though it'd be nice
